### PR TITLE
[DDW-1152] Fix general stories

### PR DESCRIPTION
**Pull request by VanessaPC**
The General stories of Storybook were either not handling the state of some props in storybook, or knobs weren't working. So now they should display the behaviour they are expected to have.
[Jira ticket](https://input-output.atlassian.net/browse/DDW-1152)

## Todos
- [x] Build storybook with ` yarn storybook:build`
- [x] Run storybook `yarn storybook`
- [x] Find the General components and check the three of them. 
- [x] Pass an story, or mock an store and pass the props appropriately

## Screenshots

Screenshot 2022-10-19 at 11 17 33
Screenshot 2022-10-19 at 11 17 40
Screenshot 2022-10-19 at 11 17 46
Screenshot 2022-10-19 at 11 17 54
Screenshot 2022-10-19 at 11 18 01
Screenshot 2022-10-19 at 11 24 23
Screenshot 2022-10-19 at 11 24 30
Screenshot 2022-10-19 at 11 24 38
Screenshot 2022-10-19 at 11 24 45
Screenshot 2022-10-19 at 11 17 28

## Testing Checklist

- [Slack QA thread](https://input-output-rnd.slack.com/messages/GGKFXSKC6)
- [ ] Test

---

- [ ] Make sure the dropdowns in General > General work 
- [ ] Make sure the switches on General > Wallets and General> Security work.

### Basics

- [x] PR assigned to the PR author(s)
- [x] `input-output-hk/daedalus-dev` and `input-output-hk/daedalus-qa` assigned as PR reviewers
- [ ] If there are UI changes, Alexander Rukin assigned as an additional reviewer
- [ ] All visual regression testing has been reviewed (assign `run Chromatic` label to PR to trigger the run)
- [x] PR has appropriate labels (`release-vNext`, `feature`/`bug`/`chore`, `WIP`)
- [x] PR link is added to a Jira ticket, ticket moved to In Review
- [x] PR is updated to the most recent version of the target branch (and there are no conflicts)
- [x] PR has a good description that summarizes all changes
- [x] PR contains screenshots (in case of UI changes)
- [x] CHANGELOG entry has been added to the top of the appropriate section (_Features_, _Fixes_, _Chores_) and is linked to the correct PR on GitHub
- [ ] There are no missing translations (running `yarn manage:translations` produces no changes)
- [ ] Text changes are proofread and approved (Jane Wild / Amy Reeve)
- [ ] Japanese text changes are proofread and approved (Junko Oda)
- [ ] Storybook works and no stories are broken (`yarn storybook`)
- [ ] In case of dependency changes `yarn.lock` file is updated

### Code Quality

- [x] Important parts of the code are properly commented and documented
- [ ] Code is properly typed with typescript types
- [ ] React components are split-up enough to avoid unnecessary re-renderings
- [ ] Any code that only works in main process is neatly separated from components

### Testing

- [ ] New feature/change is covered by acceptance tests
- [ ] New feature/change is manually tested and approved by QA team
- [ ] All existing acceptance tests are still up-to-date
- [ ] New feature/change is covered by Daedalus Testing scenario
- [ ] All existing Daedalus Testing scenarios are still up-to-date

### After Review

- [ ] Update Slack QA thread by marking it with a green checkmark

### CODE REVIEW

**Comment by gabriela-ponce** [comment]
@VanessaPC Here are a few issues/inconsistencies on this PR:

- When you click on "Security" and "Wallets", you are redirected to the tab "General". Check [this](https://drive.google.com/file/d/1QQsT0KtHGQ94d4-VVoUK9HGnetF4iVjm/view?usp=sharing) video.
- Not sure if this one is on the scope of this task, but when you enable discreet mode, the "eye" icon doesn't change (it should be crossed out), additionally, when you click on it there is an error on the screen. Check [this](https://drive.google.com/file/d/1tN0w5LU6bTo4LfKGPkX629mdGCCuckmO/view?usp=sharing) video.

File: storybook/stories/settings/general/General.stories.tsx [hunk 1]
```diff
@@ -26,6 +25,61 @@ import SecuritySettings from '../../../../source/renderer/app/components/setting
...
+      value === mockedGeneralState.currentLocale
+        ? mockedGeneralState.currentDateFormat
+        : 'YYYY年MM月DD日';
+    store.set({
+      currentDateFormat,
+    });
+  }
+};
+
 const changeControl = (name: string, value: boolean) => {
```
Comment (szymonmaslowski) [inline storybook/stories/settings/general/General.stories.tsx:83]: It looks like this guy is no longer used. Please remove it.

File: storybook/stories/settings/general/General.stories.tsx [hunk 2]
```diff
@@ -26,6 +25,61 @@ import SecuritySettings from '../../../../source/renderer/app/components/setting
...
+    name: 'unified Stable Dollar',
+  },
+};
+
+const mockedSecurityStore = {
+  discreetMode: true,
+  openDiscreetMode: false,
+};
+
+const onValueChange = (
```
Comment (szymonmaslowski) [inline storybook/stories/settings/general/General.stories.tsx:56]: It looks like the `StoryStore`, `mockedGeneralState` and the `onValueChange` are copied over from another story, right? Let's extract them to a separate file and import them in all the places ;) This file could be placed in the `stories/settings/utils` directory. I would also suggest naming them in common a way to reflect their purpose, for example: `LanguageSettingsStore`, `languageSettingsInitialState`, `onLanguageSettingsChange`.

**Review by szymonmaslowski — CHANGES_REQUESTED** [review]
Nice work! Just two small comments ;)

**Review by szymonmaslowski — APPROVED** [review]
👍

File: storybook/stories/settings/general/General.stories.tsx (hunk 2 above)
Comment (szymonmaslowski) [reply storybook/stories/settings/general/General.stories.tsx]: Right, that makes sense 🙂

**Review by dmitrii-gaico — APPROVED** [review]

**Review by miorsufianiohk — APPROVED** [review]
LGTM. Great work @VanessaPC 👍

**Review by ManusMcCole — APPROVED** [review]
👍

**Review by gabriela-ponce — APPROVED** [review]

**Pull request merged**